Assets/MixedRealityToolkit/_Core/EventDatum/Input/SourcePoseEventData.cs

```diff
@@ -32,52 +39,63 @@ public class SourcePositionEventData : SourceStateEventData
+        public SixDof SixDofPose { get; private set; } = SixDof.ZeroIdentity;
 
         /// <inheritdoc />
-        public SourcePositionEventData(EventSystem eventSystem) : base(eventSystem) { }
+        public SourcePoseEventData(EventSystem eventSystem) : base(eventSystem) { }
+
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, TrackingState trackingState)
+        {
+            Initialize(inputSource, controller);
+            TrackingState = trackingState;
+        }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector2 position)
+        /// <param name="controller"></param>
+        /// <param name="position"></param>
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector2 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             TwoDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector3 position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector3 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="rotation"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Quaternion rotation)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Quaternion rotation)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofRotation = rotation;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, SixDof position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, SixDof position)
         {
-            Initialize(inputSource);
-            SixDofPosition = position;
+            Initialize(inputSource, controller);
+            SixDofPose = position;
```

Comment:
Going back to my comments from last night:

This event data currently has 4 things it might contain:
1. `TwoDofPosition`
2. `ThreeDofPosition`
3. `ThreeDofRotation`
4. `SixDofPose`

What's the benefit of providing `SixDofPose` at all, instead of using the existing `ThreeDofPosition` and `ThreeDofRotation`. Additionally, how do I even know which value to look at?

Reply:
I didn't want to break up the interface implementation into more things users have to implement to get the data.

That way they can handle all types of source pose changes.

I'm open to splitting them up, but I didn't want a lot of interfaces for handlers.

Reply:
I think I understand and am pretty sure I agree with not wanting to split them up, but I still have those questions:

1. How do I know which values in the event data I should care about? 
2. The benefit of SixDof vs using the two ThreeDof values. This would also reduce the number of places for me to check for populated data

Reply:
You can implement all the interfaces and if you happen to get the data you need, then you process it.

Reply:
What if we made the data in the event args nullables? We can then respond to the pose changed event and check by saying:
if (twoDofData.HasValue) { … }
if (threeDofPositionData.HasValue) { … }
etc.

Reply:
Agreed. There has to be some way to actually know 

>if you happen to get the data you need

Reply:
Agreed. There has to be some way to actually know 

>if you happen to get the data you need

Reply:
I don't want to make it complicated with nullable types. I'd rather split them up into separate events before we do that.

Reply:
>>Agreed. There has to be some way to actually know

>if you happen to get the data you need

Looking back at the handler, I noticed we only have a single method call `OnSourcePoseChanged`

I propose we add specific methods to know which data types got updated:

```
OnSourceChanged(); // Gets raised for them all.
OnSource2DofPositionChanged(); // Only gets raised when a Vector2 change happens
OnSource3DofPositionChanged(); // Only gets raised when a Vector3 change happens
OnSourceRotationChanged(); // Only gets raised when a Rotation change happens
OnSourcePoseChanged(); // Only gets raised when a Pose change happens.
```

Reply:
Guys take a look at my most recent updates. This should fix a lot of the issues you guys had with this.